File: APISolidarityManager/APISolidarityManager/Models/Donation.cs
namespace APISolidarityManager.Models
{
    public class Donation
    {
        public Guid Id { get; set; }
        public Guid DonorId { get; set; }
        public DateTime ReceivedDate { get; set; }
        public Guid CreatedBy { get; set; }
        public string? Notes { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
    }
}

File: donation-system-web/src/types/donation.ts
export interface DonationItemResponse {
  inventoryBatchId: string;
  itemId: string;
  expirationDate?: string;
  quantity: number;
}

export interface DonationResponse {
  id: string;
  receivedDate: string;
  createdBy: string;
  notes?: string;
  createdAt: string;
  updatedAt?: string;
  items: DonationItemResponse[];
}

export interface DonationItemRequest {
  itemId: string;
  expirationDate?: string;
  quantity: number;
}

export interface CreateDonationRequest {
  receivedDate: string;
  notes?: string;
  items: DonationItemRequest[];
}
